src/wallet/wallet.cpp

```diff
@@ -4379,3 +4392,25 @@ void CWallet::LearnAllRelatedScripts(const CPubKey& key)
     LearnRelatedScripts(key, OutputType::P2SH_SEGWIT);
 }
 
+std::vector<OutputGroup> CWallet::GroupOutputs(const std::vector<COutput>& outputs, bool single_coin) const {
```

Comment:
Perhaps we should not create groups that would be excessively large to send together?  For instance if someone has a couple thousand outputs to the same address, then as a group they would become unspendable (because the resulting transaction would be too big).  

It might be pretty annoying for our users if they became vulnerable to weird DoS attacks where people create lots of dust to prevent them from spending their coins...

Reply:
Good point. I've set a cap at 10 outputs per group.

That being said, the code (at least BnB algo; knapsack is being deprecated) already removes dust from the group before running coin select. https://github.com/kallewoof/bitcoin/blob/19bbab013860c84475cce98a55179a1167348269/src/wallet/wallet.cpp#L2415-L2422